pkg/api/handlers/compat/images_build.go

```diff
@@ -698,12 +700,30 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 			}
 			flush()
 		case e := <-auxout.Chan():
-			m.Stream = string(e)
-			if err := enc.Encode(m); err != nil {
-				stderr.Write([]byte(err.Error()))
+			if !query.Quiet {
+				m.Stream = string(e)
+				if err := enc.Encode(m); err != nil {
+					stderr.Write([]byte(err.Error()))
+				}
+				flush()
+			} else {
+				stepErrors = append(stepErrors, string(e))
 			}
-			flush()
 		case e := <-stderr.Chan():
+			if len(stepErrors) > 0 {
```

Comment:
No need for length check. For loop will not fire if len==0.

Reply:
fixed :)